src/snowflake/cli/plugins/nativeapp/artifacts.py

```diff
@@ -75,37 +60,300 @@ class NotInDeployRootError(ClickException):
     (use "./" instead to copy into the deploy root).
     """
 
-    artifact_src: str
```

Comment:
My guess is that the `artifact_src` was there just for an FYI, so a user will know the source to the file that should have been there in the deploy root. Do we not want to provide that information anyway? Maybe having the artifact src in there helps the user narrow the error faster in snowflake.yml, as opposed to having multiple artifact src's end up in the same dest path and combing through each of them to figure out why it threw an error?

Reply:
The issue is I don't always have a source given the abstractions in place, and it would make the code really tricky if we had to thread it through in all cases. I added it back as an optional piece, on a best effort basis. Does that work?

Reply:
That works!